### DISCUSSION TITLE
onViewableItemsChanged - Flashing behavior

### DISCUSSION THREAD

**Discussion by HassenMabrouki** [Shopify/flash-list#583] [Q&A]
I use flashlist to render Items in fullScreen, and I use onViewableItemsChanged to get the item.id of the currently visible item on the screen (videos show as full screen), but it causes a flashing behavior. Any idea how to prevent this, please?

## FlashList component:
```
<View style={styles.container}>
        <FlashList
          extraData={visibleSlideId}
          data={data}
          decelerationRate={'fast'}
          estimatedItemSize={height}
          keyExtractor={item => item.id.toString()}
          onViewableItemsChanged={visibleSlideIdRef.current}
          ref={mainScroll}
          renderItem={({ item, extraData, index }) => <RowRenderer item={item} visibleSlideId={extraData} index={index} />}
          snapToInterval={height}
          viewabilityConfig={{ viewAreaCoveragePercentThreshold: 80 }}
        />
    </View>
```
## Video:
https://user-images.githubusercontent.com/95028333/188112702-d68a9621-ec95-4063-b1be-27c9a3ba574c.mov

**Comment by fortmarek** [comment]
Hey @HassenMabrouki

`onViewableItemsChanged` should not impact how view is rendered. From the snippet, I can't also tell why that would be the case. Could you provide a full reproducible sample instead? Once you do so, we can also migrate this to an issue instead 👍
